### PR TITLE
Fix clippy warnings

### PR DESCRIPTION
**Pull request by ishitatsuyuki**
Also, changed try! to ? operator, and removed some nonsense occurrences.

... please learn about style.

### CODE REVIEW

**Comment by ishitatsuyuki** [comment]
I removed the change to `ops::intersect_packages` which makes it take a slice instead of Vec, since that will break API compatibility. Do it at your risk and bump the version if doing so.

File: src/main-config.rs [hunk 1]
```diff
@@ -26,12 +26,12 @@ fn actual_main() -> Result<(), i32> {
             configuration.insert(opts.package.clone(), cargo_update::ops::PackageConfig::from(&opts.ops));
         }
 
-        try!(cargo_update::ops::PackageConfig::write(&configuration, &config_file));
+        cargo_update::ops::PackageConfig::write(&configuration, &config_file)?;
```
Comment (nabijaczleweli) [inline src/main-config.rs]: `?` -> `try!()`

File: src/main-config.rs [hunk 2]
```diff
@@ -15,7 +15,7 @@ fn actual_main() -> Result<(), i32> {
     let opts = cargo_update::ConfigOptions::parse();
     let config_file = cargo_update::ops::resolve_crates_file(opts.crates_file.1).with_file_name(".install_config.toml");
 
-    let mut configuration = try!(cargo_update::ops::PackageConfig::read(&config_file));
+    let mut configuration = cargo_update::ops::PackageConfig::read(&config_file)?;
```
Comment (nabijaczleweli) [inline src/main-config.rs]: `?` -> `try!()`

File: src/main.rs [hunk 3]
```diff
@@ -31,7 +31,7 @@ fn actual_main() -> Result<(), i32> {
     }
 
     let crates_file = cargo_update::ops::resolve_crates_file(opts.crates_file.1);
-    let configuration = try!(cargo_update::ops::PackageConfig::read(&crates_file.with_file_name(".install_config.toml")));
+    let configuration = cargo_update::ops::PackageConfig::read(&crates_file.with_file_name(".install_config.toml"))?;
```
Comment (nabijaczleweli) [inline src/main.rs]: `?` -> `try!()`

File: src/ops/config.rs [hunk 4]
```diff
@@ -201,9 +201,9 @@ impl PackageConfig {
     pub fn read(p: &Path) -> Result<BTreeMap<String, PackageConfig>, i32> {
         if p.exists() {
             let mut buf = String::new();
-            try!(try!(File::open(p).map_err(|_| 1))
+            File::open(p).map_err(|_| 1)?
```
Comment (nabijaczleweli) [inline src/ops/config.rs]: `?` -> `try!()`

File: src/ops/config.rs [hunk 5]
```diff
@@ -201,9 +201,9 @@ impl PackageConfig {
...
                 .read_to_string(&mut buf)
-                .map_err(|_| 1));
+                .map_err(|_| 1)?;
```
Comment (nabijaczleweli) [inline src/ops/config.rs]: `?` -> `try!()`

File: src/ops/config.rs [hunk 6]
```diff
@@ -245,8 +245,8 @@ impl PackageConfig {
     ///                   features = [\"serde\"]\n");
     /// ```
     pub fn write(configuration: &BTreeMap<String, PackageConfig>, p: &Path) -> Result<(), i32> {
-        try!(File::create(p).map_err(|_| 3))
-            .write_all(&try!(toml::to_vec(configuration).map_err(|_| 2)))
+        File::create(p).map_err(|_| 3)?
```
Comment (nabijaczleweli) [inline src/ops/config.rs]: `?` -> `try!()`

File: src/ops/config.rs [hunk 7]
```diff
@@ -245,8 +245,8 @@ impl PackageConfig {
...
+            .write_all(&toml::to_vec(configuration).map_err(|_| 2)?)
```
Comment (nabijaczleweli) [inline src/ops/config.rs]: `?` -> `try!()`

File: src/ops/config.rs [hunk 8]
```diff
@@ -150,15 +150,15 @@ impl PackageConfig {
     ///            });
     /// ```
     pub fn execute_operations<'o, O: IntoIterator<Item = &'o ConfigOperation>>(&mut self, ops: O) {
-        for op in ops.into_iter() {
-            match op {
-                &ConfigOperation::SetToolchain(ref tchn) => self.toolchain = Some(tchn.clone()),
-                &ConfigOperation::RemoveToolchain => self.toolchain = None,
-                &ConfigOperation::DefaultFeatures(f) => self.default_features = f,
-                &ConfigOperation::AddFeature(ref feat) => {
+        for op in ops {
```
Comment (nabijaczleweli) [inline src/ops/config.rs]: Cool, didn't know it autodecayed.

**Comment by nabijaczleweli** [comment]
Also rebase upon current `master` while you're at it.

**Comment by ishitatsuyuki** [comment]
Reverted try! changes as mentioned. Also did a rebase, although I don't see there's any point to do that (no merge conflicts, and you seems to use the rebase merge button).

**Comment by nabijaczleweli** [comment]
I don't use buttons.

**Comment by nabijaczleweli** [comment]
Released in [`v0.8.2`](https://github.com/nabijaczleweli/cargo-update/releases/tag/v0.8.2).

**Pull request merged**